cli/src/description_util.rs

```diff
@@ -40,6 +45,110 @@ JJ: Lines starting with "JJ: " (like this one) will be removed.
     Ok(text_util::complete_newline(description.trim_matches('\n')))
 }
 
+#[derive(Debug)]
+pub struct EditMultipleDescriptionsResult {
+    /// The parsed, formatted descriptions.
+    pub descriptions: HashMap<CommitId, String>,
+    /// Commit IDs that were found multiple times while parsing the edited
+    /// messages.
+    pub duplicates: Vec<String>,
+    /// Commit IDs that were found while parsing the edited messages, but which
+    /// were not originally being edited.
+    pub unexpected: Vec<String>,
+}
+
+/// Edits the descriptions of the given commits in a single editor session.
+pub fn edit_multiple_descriptions(
+    ui: &Ui,
+    settings: &UserSettings,
+    workspace_command: &WorkspaceCommandHelper,
+    repo: &ReadonlyRepo,
+    commits: &[Commit],
+) -> Result<EditMultipleDescriptionsResult, CommandError> {
+    let id_prefix_context = workspace_command.id_prefix_context()?;
+    let mut commits_map = HashMap::new();
+    let mut output_chunks = Vec::new();
+
+    for commit in commits.iter().rev() {
```

Comment:
nit: maybe better to reverse `commits` by caller? it's unclear why this function do `.rev()`.